backend/onyx/agents/agent_search/shared_graph_utils/utils.py

```
@@ -372,7 +376,20 @@ def summarize_history(
         )
     )
 
-    history_response = llm.invoke(history_context_prompt)
+    try:
+        history_response = llm.invoke(
+            history_context_prompt,
+            timeout_overwrite=AGENT_TIMEOUT_OVERWRITE_LLM_HISTORY_SUMMARY_GENERATION,
+        )
+    except openai.APITimeoutError:
+        return (
+            history  # this is what is done at this point anyway, so wwe default to this
+        )
+
+    except Exception:
```

I wonder if it's better to fail loudly and propagate up the stack if not an exception we are explicitly handling. Here and in other places. It prevents silent failures from staying around for a long time

---

In general, I agree. Here however it is only a nice to have.

---

fwiw except Exception make me very sad :'(

---

DONE!!